packages/twenty-front/src/modules/views/view-picker/components/ViewPickerContentCreateMode.tsx

```diff
@@ -92,6 +92,10 @@ export const ViewPickerContentCreateMode = () => {
   const defaultIcon =
     viewPickerType === ViewType.Kanban ? 'IconLayoutKanban' : 'IconTable';
 
+  if (!hasManuallySelectedIcon) {
+    setViewPickerSelectedIcon(defaultIcon);
+  }
```

Comment:
🤔

Reply:
I get the concern about state updates in render body - it's usually a red flag. But in this case, it actually works correctly because hasManuallySelectedIcon prevents any infinite loop. The update only happens once during initial render when no icon is manually picked, and then that's it. I tried moving it to an effect but that broke the expected behavior where we need the default icon right away for both display and save logic. Happy to explore other approaches if you see potential issues I might have missed though!

Reply:
I agree with the bot, this can lead to a lot of calls to `setViewPickerSelectedIcon`.
Other re-renders can be triggered by other states or components, and if `hasManuallySelectedIcon` is already at false it's going to set the picker selected icon again.
You should definitely put this in an `useEffect`, you can also try with `useLayoutEffect` if you need to set it before the dom painting.

Reply:
I tried both useEffect and useLayoutEffect, but unfortunately, neither works. The current approach ensures the icon state is available right when it's needed. Let me know if you have other ideas for this! Weird bug :)